processor/spanmetricsprocessor/processor.go

```diff
@@ -529,6 +566,7 @@ func getDimensionValue(d Dimension, spanAttr pdata.AttributeMap, resourceAttr pd
 //   LabelsMap().InitFromMap(p.metricKeyToDimensions[key])
 func (p *processorImp) cache(serviceName string, span pdata.Span, k metricKey, resourceAttrs pdata.AttributeMap) {
 	p.metricKeyToDimensions.ContainsOrAdd(k, p.buildDimensionKVs(serviceName, span, p.dimensions, resourceAttrs))
+	stats.Record(context.TODO(), dimensionsCacheSize.M(int64(p.metricKeyToDimensions.Len())))
```

Comment:
Same question as the `uniqueTimeSeries` metric, can this be done outside of the loop